pyproject.toml

```diff
@@ -369,7 +369,7 @@ furo = ">=2022.12.7,<2024.0.0"
 # system changes.
 # We are happy to raise these upper bounds upon request,
 # provided we check that it's safe to do so (i.e. that CI passes).
-requires = ["poetry-core>=1.1.0,<=1.6.0", "setuptools_rust>=1.3,<=1.6.0"]
+requires = ["poetry-core>=1.1.0,<1.7", "setuptools_rust>=1.3,<1.7"]
```

Comment:
This is pedantic I know, but semantic versioning would be `<2` for both, since the minor releases are supposedly only meant to add backwards-compatible features.

The problem is that we've been burned by this (minor releases with breaking changes) in the past.

I think I'm with @DMRobertson here in that I'd rather keep these bounds defensive.
If we don't and the build system maintainers make a breaking release, this breaks the ability to easily install old PyPI packages or even old dependency-locked packages of Synapse from the git repository (because the build system is itself not locked, except by these bounds here).

I understand the pain this causes for packagers and sorry for that, but there isn't really a perfect choice here. (We're very happy for packagers to patch/sed the bounds up to fit your packaging situation in exchange, as well as to bump up the limits when released packages emerge that we can test in CI.)